File: kotlin/lib/src/main/kotlin/models/BackgroundTaskType.kt
// This file is @generated
package com.svix.kotlin.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.encodeToJsonElement
import kotlinx.serialization.json.jsonPrimitive

@Serializable
public enum class BackgroundTaskType {
    @SerialName("endpoint.replay") ENDPOINT_REPLAY,
    @SerialName("endpoint.recover") ENDPOINT_RECOVER,
    @SerialName("application.stats") APPLICATION_STATS,
    @SerialName("message.broadcast") MESSAGE_BROADCAST,
    @SerialName("sdk.generate") SDK_GENERATE,
    @SerialName("event-type.aggregate") EVENT_TYPE_AGGREGATE;

    override fun toString(): String = Json.encodeToJsonElement(this).jsonPrimitive.content
}

File: javascript/src/models/background_task_type.ts
// this file is @generated

export enum BackgroundTaskType {
  EndpointReplay = "endpoint.replay",
  EndpointRecover = "endpoint.recover",
  ApplicationStats = "application.stats",
  MessageBroadcast = "message.broadcast",
  SdkGenerate = "sdk.generate",
  EventTypeAggregate = "event-type.aggregate",
}

export namespace BackgroundTaskTypeUtil {
  export function _fromJsonObject(object: any): BackgroundTaskType {
    return object;
  }

  export function _toJsonObject(self: BackgroundTaskType): any {
    return self;
  }
}
